### PR TITLE
0156 sre server

### PR DESCRIPTION
**Pull request by mrkooll**
Automation part.

All automation is driven by make:

* `make cluster` (or just `make`)
    - build binary
    - build docker image
    - start kind cluster
    - generate CA and server certificate
    - upload docker image and certificates to the cluster
    - build helm chart
    - deploy helm chart
 * `make test`
    - generate client certificate
    - run shell script with tests
 * `make clean`
    - stop and delete cluster
    - remove all generated files

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FROM alpine:3.19.0
```
Comment (jimbishopp) [inline Dockerfile]: Does this need alpine? Can it use a distroless image instead?

File: sre-server/test.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+PORT=8443
+
+CURL="curl -v --cert ${WD}/client.crt --key ${WD}/client.key --cacert ${WD}/ca.crt -H \"Content-Type: application/json\""
+
+PASSED () {
+    echo -e "\033[1m\033[42mPASSED\033[0m"
+}
+FAILED () {
+    echo -e "\033[1m\033[41mFAILED\033[0m"
+}
```
Comment (jimbishopp) [inline sre-server/test.sh]: Should this exit on failure?

**Review by jimbishopp — APPROVED** [review]
Thanks for submitting this so quickly! A couple minor comments.

**Comment by mrkooll** [comment]
> Thanks for submitting this so quickly! A couple minor comments.

As I remember Jan 03 is my deadline. Since my sickness is not your problem I created this PR as fast as I can.

File: Dockerfile (hunk 1 above)
Comment (mrkooll) [reply Dockerfile]: To be honest, I don't need alpine here. I need shell only.
Changed to busybox to have shell support.

File: sre-server/test.sh (hunk 2 above)
Comment (mrkooll) [reply sre-server/test.sh]: I don't think that exit after first failed test is good practice. Usually test packages always run all tests.

I added exit code and test.sh will exit with code 1 when any test will fail.

**Comment by mrkooll** [comment]
Updated according to the review's comments.

**Review by sclevine — APPROVED** [review]

**Review by jimbishopp — APPROVED** [review]

**Pull request merged**